core/client/src/main/java/alluxio/client/file/FileInStream.java

```diff
@@ -232,6 +238,50 @@ public int read(byte[] b, int off, int len) throws IOException {
   }
 
   @Override
+  public int positionedRead(long pos, byte[] b, int off, int len) throws IOException {
+    if (!PACKET_STREAMING_ENABLED) {
+      throw new RuntimeException(String.format(
+          "Positioned read is not supported, please set %s to true to enable positioned read.",
+          PropertyKey.USER_PACKET_STREAMING_ENABLED.toString()));
+    }
+    if (pos < 0 || pos >= mFileLength) {
+      return -1;
+    }
+
+    // If partial read cache is enabled, we fall back to the normal read.
+    if (mShouldCachePartiallyReadBlock) {
```

Comment:
If the user does a positioned read of a complete block, it will not be cached if partial caching is turned off right?